tests/rustdoc/extern/extern-html-alias.rs

```diff
@@ -0,0 +1,8 @@
+//@ compile-flags:-Z unstable-options --extern-html-root-url extern_prelude_name=https://renamed.example.com  --extern-html-root-url empty=https://bad.invalid
+//@ aux-crate:extern_prelude_name=empty.rs
+
+extern crate extern_prelude_name as internal_ident_name;
+
+//@ has extern_html_alias/index.html
+//@ has - '//a/@href' 'https://renamed.example.com/empty/index.html'
+pub use internal_ident_name as yet_different_name;
```

Comment:
1. `internal_ident_name` is a misleading name, as it actually does exist in extern prelude as an alias (i.e. `::internal_ident_name` would be valid here, in this case both exist in the extern prelude)
2. we should test that `extern crate as` aliases are handled correctly
3. we should test the fallback behavior (using crate name)

Doing all this would probably require at least a second `aux-crate`.

Reply:
`Resolver.extern_prelude` confused me, because it contains only `core`, `std`, and `extern_prelude_name`, and not `internal_ident_name`. 

I've renamed things to avoid referring to the extern prelude.